### DISCUSSION TITLE
Prompt for LLM

### DISCUSSION THREAD

**Discussion by Arcovoltaire** [reorproject/reor#94] [Ideas]
I would like to sincerely ask you to provide a way to change the prompt sent with the context, so that I can say in which language the LLM should respond and in what tone, and commands similar to that.

**Comment by samlhuillier** [comment]
Yes this is something I can get to this week. Fairly straightforward task. Would you be interested in opening a PR for this perhaps?

**Comment by Arcovoltaire** [comment]
Great!! Look, I'm a regular user who saw a video on YouTube and downloaded the program, I don't know what "opening a PR" means. But I think the program is very promising, I'm testing it. I arranged to use it with lmstudio and so it was faster than using lm natively. I tested some embeds. One difficulty I have is that it is slow with large texts, the embed processes a lot. Especially when we try to make a change. I know that the embedes are the core of the program and I love them. I always see what they are sending to llm through lmstudio. I am from Brazil.

**Comment by samlhuillier** [comment]
Ah gotcha lol. A PR is a pull request which is how contributions are made to an open source project :)

What kind of hardware do you have on your computer? Trying to understand why it is slow...

**Comment by Arcovoltaire** [comment]
Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz - Ram 16g - GPU GTX 1050
LLM LOCAL via LMStudio - mistral-7b-instruct-v0.2.Q4_K_S.gguf
Embedding Model: Xenova/multilingual-e5-large
Large text size: Pasting as if it were 5 to 7 pages of Word.
Waiting time to process: Around 5 minutes

**Comment by Arcovoltaire** [comment]
For now, so I can continue testing, is there any way for me to access the prompt through vscode?

**Comment by Arcovoltaire** [comment]
I managed to run as a developer, and accessed the prompt inside the electron folder. I changed it and it's working very well for my language now.
![tempsnip](https://github.com/reorproject/reor/assets/111749224/ea0bfd3d-73d9-4071-bb36-78ae13e177a6)

**Comment by Arcovoltaire** [comment]
LLm is now much more in tune with my purpose and responding in Portuguese.
![tempsnip](https://github.com/reorproject/reor/assets/111749224/8b6ca533-6867-4b34-b5a4-abb4a62ad380)

**Comment by samlhuillier** [comment]
@Arcovoltaire interesting this seems to make a big difference! I'm working on making this customisable this week in the app. Our interfaces overall to the LLMs are not the best and there are lots of places to make improvements. I'll keep you posted :)

**Comment by BradKML** [comment]
We need multiple ways of doing prompts (including AutoPrompts that are optimize for learning) https://github.com/Eladlev/AutoPrompt

- Writing and prose aid (to be mixed with prose linters)
- Research and summarization agents (to be mixed with crawlers and document parsers)
- Socratic dialogue agent (to be mixed with multi-agent systems to argue for different POVs)
- (don't even want to start with code agents like OpenHands or experiment agents like SakanaAI otherwise it will be too OP)

See also the one-man band work @samlhuillier cus that is on the verge of being impressive https://github.com/rmusser01/tldw
